### PR TITLE
refactor: move pwa/build-info -> appConfig

### PR DESCRIPTION
**Pull request by danielroe**
This is a slight performance optimisation. We likely don't want to turn on/off pwa being enabled (or update build info) at runtime. So by moving this into app config (which is built into code) we both save the payload and enable the possibility of tree shaking.

This is a precursor to tree-shaking out different unstorage drivers (but this would be a breaking change so will defer to a different PR).

### CODE REVIEW

File: modules/pwa/index.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ export default defineNuxtModule<VitePWANuxtOptions>({
     }
     let webmanifests: LocalizedWebManifest | undefined
 
+    nuxt.options.appConfig = nuxt.options.appConfig || {}
+    nuxt.options.appConfig.pwaEnabled = !nuxt.options.dev || process.env.VITE_DEV_PWA === 'true'
```
Comment (userquin) [inline modules/pwa/index.ts:27]: how about tauri, rn it is disabled

Comment (danielroe) [reply modules/pwa/index.ts:27]: This PR doesn't change the current behaviour, just moves the setting from `nuxt.config` -> `pwa/index`. Do you think we need to change the current behaviour - or am I missing something?

Comment (userquin) [reply modules/pwa/index.ts:27]: `nuxt.options.appConfig.pwaEnabled = !nuxt.options.dev || process.env.VITE_DEV_PWA === 'true'` that's true also on Tauri

Comment (userquin) [reply modules/pwa/index.ts:27]: I mean, `!nuxt.options.dev`

Comment (danielroe) [reply modules/pwa/index.ts:27]: Current login is the same: https://github.com/elk-zone/elk/blob/main/nuxt.config.ts#L101

**Review by antfu — APPROVED** [review]

**Comment by danielroe** [comment]
One further change coming (discussing with @userquin).

**Review by userquin — APPROVED** [review]

**Pull request merged**